### PR TITLE
Add mypy for `protocol.invite`

### PR DESCRIPTION
**Pull request by Volham22**
This PR enable mypy for `parsec.protocol.invite.*`

### CODE REVIEW

File: parsec/core/backend_connection/authenticated.py [hunk 1]
```diff
@@ -69,23 +143,31 @@ def __init__(self, addr: BackendOrganizationAddr, acquire_transport):
     ping = expose_cmds_with_retrier(cmds.authenticated_ping)
     message_get = expose_cmds_with_retrier(cmds.message_get)
     user_get = expose_cmds_with_retrier(cmds.user_get)
-    user_create = expose_cmds_with_retrier(cmds.user_create)
+    user_create: UserCreateCallbackType = expose_cmds_with_retrier(cmds.user_create)
     user_revoke = expose_cmds_with_retrier(cmds.user_revoke)
-    device_create = expose_cmds_with_retrier(cmds.device_create)
+    device_create: DeviceCreateCallbackType = expose_cmds_with_retrier(cmds.device_create)
```
Comment (vxgmichel) [inline parsec/core/backend_connection/authenticated.py:148]: Why do some commands need to be typed and not others?

**Review by vxgmichel — APPROVED** [review]
One question, LGTM otherwise :+1:

File: parsec/core/backend_connection/authenticated.py [hunk 2]
```diff
@@ -59,6 +71,68 @@ def copy_exception(exception: BaseExceptionTypeVar) -> BaseExceptionTypeVar:
     return result
 
 
+# Authenticated commands types
+class DeviceCreateCallbackType(Protocol):
+    def __call__(
+        self, device_certificate: bytes, redacted_device_certificate: bytes
+    ) -> Awaitable[DeviceCreateRep]:
+        ...
```
Comment (TimeEngineer) [inline parsec/core/backend_connection/authenticated.py:79]: Is it normal to create empty `def` in python file ?

Comment (Volham22) [reply parsec/core/backend_connection/authenticated.py:79]: I'll remove theses protocol once #3325 merged. They're not needed if `expose_cmds` and `expose_cmds_with_retrier` are properly typed which is done by #3325 (these functions are parts of `core.backend_connection`).

**Review by TimeEngineer — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Review by vxgmichel — APPROVED** [review]

**Pull request merged**